File: src/main/java/mediathek/gui/actions/StartDownloadsAction.java
package mediathek.gui.actions;

import mediathek.gui.tabs.tab_downloads.GuiDownloads;
import mediathek.tool.SVGIconUtilities;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class StartDownloadsAction extends AbstractAction {
    private final GuiDownloads guiDownloads;

    public StartDownloadsAction(GuiDownloads guiDownloads) {
        this.guiDownloads = guiDownloads;
        putValue(Action.NAME, "Ausgewählte Downloads starten");
        putValue(Action.SMALL_ICON, SVGIconUtilities.createSVGIcon("icons/fontawesome/caret-down.svg"));
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        guiDownloads.starten(false);
    }
}

File: src/main/kotlin/mediathek/gui/actions/StartDownloadsAction.kt
package mediathek.gui.actions

import mediathek.gui.tabs.tab_downloads.GuiDownloads
import mediathek.tool.SVGIconUtilities
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class StartDownloadsAction(private val guiDownloads: GuiDownloads) : AbstractAction() {
    init {
        putValue(NAME, "Ausgewählte Downloads starten")
        putValue(SMALL_ICON, SVGIconUtilities.createSVGIcon("icons/fontawesome/caret-down.svg"))
    }

    override fun actionPerformed(e: ActionEvent?) {
        guiDownloads.starten(false)
    }
}
